From 5ef5b1ca28acf681f5097196ea573c42e85b6f8e Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Tue, 17 Mar 2026 23:23:05 +0000
Subject: [PATCH 2/2] Lower minimum kernel version requirements for GSI
 compatibility

Lower q/android-4.9 minlts from 4.9.165 to 4.9.141 and remove
CONFIG_XFRM_INTERFACE and USB/RTL8152 dependencies that are not
available on older kernel point releases.

Lower s/android-4.19 minlts from 4.19.191 to 4.19.127 for MediaTek
devices that ship 4.19 kernels at earlier point releases.

Change-Id: I08e75538ac24c48cf818f599e4aed73afeb3af40
---
 q/android-4.9/android-base-conditional.xml  | 16 +---------------
 q/android-4.9/android-base.config           |  1 -
 s/android-4.19/android-base-conditional.xml |  2 +-
 3 files changed, 2 insertions(+), 17 deletions(-)

diff --git a/q/android-4.9/android-base-conditional.xml b/q/android-4.9/android-base-conditional.xml
index 0ba7e5b..ba664a7 100644
--- a/q/android-4.9/android-base-conditional.xml
+++ b/q/android-4.9/android-base-conditional.xml
@@ -1,4 +1,4 @@
-<kernel minlts="4.9.165" />
+<kernel minlts="4.9.141" />
 
 <!-- KEEP ALPHABETICALLY SORTED -->
 <!-- ARM base requirements -->
@@ -143,17 +143,3 @@
 	</config>
 </group>
 
-<!-- CONFIG_USB implies CONFIG_USB_RTL8152 -->
-<group>
-	<conditions>
-		<config>
-			<key>CONFIG_USB_RTL8152</key>
-			<value type="bool">n</value>
-		</config>
-	</conditions>
-	<config>
-		<key>CONFIG_USB</key>
-		<value type="bool">n</value>
-	</config>
-</group>
-
diff --git a/q/android-4.9/android-base.config b/q/android-4.9/android-base.config
index 795cde5..136c2c2 100644
--- a/q/android-4.9/android-base.config
+++ b/q/android-4.9/android-base.config
@@ -218,6 +218,5 @@ CONFIG_USB_CONFIGFS_F_MIDI=y
 CONFIG_USB_CONFIGFS_UEVENT=y
 CONFIG_USB_GADGET=y
 CONFIG_USB_SUPPORT=y
-CONFIG_XFRM_INTERFACE=y
 CONFIG_XFRM_STATISTICS=y
 CONFIG_XFRM_USER=y
diff --git a/s/android-4.19/android-base-conditional.xml b/s/android-4.19/android-base-conditional.xml
index 8aab2a0..24e2b47 100644
--- a/s/android-4.19/android-base-conditional.xml
+++ b/s/android-4.19/android-base-conditional.xml
@@ -1,4 +1,4 @@
-<kernel minlts="4.19.191" />
+<kernel minlts="4.19.127" />
 
 <!-- KEEP ALPHABETICALLY SORTED -->
 <!-- ARM base requirements -->
-- 
2.53.0

